CRM/Core/Payment/PayPalImpl.php

```diff
@@ -687,9 +687,20 @@ public function cancelSubscription(&$message = '', $params = array()) {
    * We don't need to handle paypal standard using this path as there has never been any historic support
    * for paypal standard to call civicrm/payment/ipn as a path.
    */
-  static public function handlePaymentNotification() {
-    $paypalIPN = new CRM_Core_Payment_PayPalProIPN($_REQUEST);
```

Comment:
We can't be 1000% sure that there are no processors out there calling this function for paypal Pro - it does work in theory...

However if we are using civicrm/payment/ipn/xx url then the xx will be the payment processor id & using that we can be 100% sure about whether it is paypal Pro or Std